src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotDecryptWithoutHashTests.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing;
+using Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.NetCore.Analyzers.Security.UnitTests
+{
+    public class DoNotDecryptWithoutHashTests : TaintedDataAnalyzerTestBase
+    {
+        public DoNotDecryptWithoutHashTests(ITestOutputHelper output)
+            : base(output)
+        {
+        }
+
+        protected override DiagnosticDescriptor Rule => DoNotDecryptWithoutHash.Rule;
+
+        [Fact]
+        public void Test_DecryptByteArray_Diagnostic()
+        {
+            VerifyCSharp(@"
+using System.IO;
+using System.Security.Cryptography;
+class TestClass
+{
+    public void TestMethod(byte[] buffer, int offset, int count, Stream stream, CryptoStreamMode mode)
+    {
+        var decryptor = new AesCng().CreateDecryptor(); 
+        var cryptoStream = new CryptoStream(stream, decryptor, mode);
+        cryptoStream.Write(buffer, offset, count);
+    }
+}",
+            GetCSharpResultAt(10, 9, 6, 28, "void CryptoStream.Write(byte[] buffer, int offset, int count)", "void TestClass.TestMethod(byte[] buffer, int offset, int count, Stream stream, CryptoStreamMode mode)", "byte[] buffer", "void TestClass.TestMethod(byte[] buffer, int offset, int count, Stream stream, CryptoStreamMode mode)"));
+        }
+
+        [Fact]
+        public void Test_LocalVariableByteArray_Diagnostic()
+        {
+            VerifyCSharp(@"
+using System.IO;
+using System.Security.Cryptography;
+class TestClass
+{
+    public void TestMethod(int offset, int count, Stream stream, CryptoStreamMode mode)
+    {
+        var byteArray = new byte[count];
+        var decryptor = new AesCng().CreateDecryptor(); 
+        var cryptoStream = new CryptoStream(stream, decryptor, mode);
+        cryptoStream.Write(byteArray, offset, count);
+    }
+}",
+            GetCSharpResultAt(11, 9, 8, 13, "void CryptoStream.Write(byte[] buffer, int offset, int count)", "void TestClass.TestMethod(int offset, int count, Stream stream, CryptoStreamMode mode)", "byte[] byteArray", "void TestClass.TestMethod(int offset, int count, Stream stream, CryptoStreamMode mode)"));
+        }
+
+        [Fact]
+        public void Test_ByteArrayFromCallingAMethod_Diagnostic()
+        {
+            VerifyCSharp(@"
+using System.IO;
+using System.Security.Cryptography;
+class TestClass
+{
+    public byte[] GetArray(int count)
+    {
```

Comment:
Would also be good to see unit tests for:
- a method that has an (out byte[] array) parameter.
- a method that populates a passed in array, e.g. Read(byte[] buffer)

And those cases with computing a hash.

Reply:
For the second case, it won't produce diagnostic cause I only taint array that in methods specified in source info and untaint array specified in sanitizer info explicitly. 
Do we need to flag it as diagnostic? Please take a look at the new added test case Test_PopulateAPassedInByteArray_NoDiagnostic.

Reply:
I was thinking along the lines of:
```csharp
byte[] bytes = new byte[1024 * 64];
SomethingReads(bytes, 0, bytes.Length);
// decrypt bytes
```

Tests that also use MemoryBuffer.ToArray() and MemoryBuffer.GetBuffer() would also be good.